framework/src/util/Properties.cpp

```diff
@@ -24,100 +24,105 @@
 
 #include <limits>
 #include <stdexcept>
-#ifdef US_PLATFORM_WINDOWS
-#  include <string.h>
-#  define ci_compare strnicmp
-#else
-#  include <strings.h>
-#  define ci_compare strncasecmp
-#endif
+
+#include "PropsCheck.h"
 
 US_MSVC_PUSH_DISABLE_WARNING(4996)
 
 namespace cppmicroservices {
 
 const Any Properties::emptyAny;
 
+Properties::Properties(AnyMap& p)
+  : props(p)
+{
+  if (p.size() > static_cast<std::size_t>(std::numeric_limits<int>::max())) {
```

Comment:
Just wondering if this check is needed. p is a reference to an AnyMap which already has a size limit.

Reply:
Fixed.